xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
I tried to look around how other parts of the Kodi code do it, but none of those quite 'fit' - might also be because a large part of the code base predates C++11. I agree that it would be better to find a set of common models that the code uses consistently, if only for new code in the beginning. I'll just move this to the Wayland folder for now. If you have concrete ideas on how the "future" should look like we can try them out here though :-)